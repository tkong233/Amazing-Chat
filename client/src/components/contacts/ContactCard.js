import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from "react-redux";
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import Avatar from '@material-ui/core/Avatar';
import ExpandLess from '@material-ui/icons/ExpandLess';
import ExpandMore from '@material-ui/icons/ExpandMore';
import Collapse from '@material-ui/core/Collapse';
import DeleteIcon from '@material-ui/icons/Delete';
import { deleteContact } from '../../actions/contactActions';
import { joinRoom, loadPastMessages, receiveNewMessages, receiveVideoCall } from '../../actions/chatActions';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    maxWidth: '25ch',
    backgroundColor: theme.palette.background.paper,
  },
  inline: {
    display: 'inline',
  },
  button: {
    maxWidth: '5px'
  }
}));

const transformDateFormat = (date) => {
  if (date) {
    const newDate = (new Date(date)).toString();
    return newDate.substring(15, 21) + ', ' + newDate.substring(3, 11);
  }
  return "";
}

const ContactCard = (props) => {
  const { name, profilePicture, lastInteractTime, userEmail, contactEmail } = props;
  const socket = props.chat.socket;
  console.log(socket);
  const classes = useStyles();
  const [open, setOpen] = useState(false);
  const toggleCollapse = () => {
    setOpen(!open);
  };

  const launchChat = () => {
    if (props.chat.pairId === props.pairId) {
      socket.room = props.pairId;
      console.log('already joined room');
      return;
    }

    props.joinRoom(props.user.name, name, userEmail, contactEmail, props.pairId, socket);
    props.loadPastMessages(props.pairId);
    props.receiveNewMessages(socket);
    props.receiveVideoCall(socket, userEmail);
  }

  return (
    <div data-test = "ContactCardComponent">
        <ListItem onClick={launchChat}>
          <ListItemAvatar>
            <Avatar alt={name} src={profilePicture}/>
          </ListItemAvatar>
          <ListItemText
          primary={name}
          secondary={transformDateFormat(lastInteractTime)}
          />
        {open ? <ExpandLess onClick={toggleCollapse}/> : <ExpandMore onClick={toggleCollapse}/>}
        </ListItem>
        <Collapse in={open} timeout="auto" unmountOnExit>
        <List component="div" disablePadding>
          <ListItem
            button
            className={classes.nested}
            onClick={() => {
              props.deleteContact(userEmail, contactEmail);
              toggleCollapse();
            }} >
          <ListItemIcon>
              <DeleteIcon />
            </ListItemIcon>
            <ListItemText primary="Delete" />
          </ListItem>
        </List>
      </Collapse>
      </div>
  );
}

const mapStateToProps = state => ({
  user: state.auth.user,
  chat: state.chat
});

export default connect(
  mapStateToProps, 
  {
    deleteContact,
    joinRoom,
    loadPastMessages,
    receiveNewMessages,
    receiveVideoCall,
  }
)(ContactCard);