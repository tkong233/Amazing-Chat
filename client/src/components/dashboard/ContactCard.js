import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
  root: {
    minWidth: 275,
    maxWidth: 300,
    textAlign: 'center'
  },
  avatar: {
    marginLeft: 'auto',
    marginRight: 'auto',
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
});

const ContactCard = (props) => {
  const classes = useStyles();
  return (
    <Card className={classes.root}>
    <CardContent>
        <Avatar alt={props.name} src={props.picture} className={classes.avatar}/>
        <Typography variant="h6" component="h3">
          {props.name}
        </Typography>
        <Button size="small">Add</Button>
      </CardContent>
    </Card>
  );
}

export default ContactCard;