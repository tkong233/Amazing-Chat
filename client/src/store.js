import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { persistStore, persistReducer } from 'redux-persist'
import rootReducer from "./reducers";
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web

const initialState = {};

const persistConfig = {
  key: 'root',
  storage,
}

const middleware = [thunk];

const persistedReducer = persistReducer(persistConfig, rootReducer, initialState);

const store = createStore(

  persistedReducer,
  compose(
    applyMiddleware(...middleware),
    (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()) ||
      compose
  )
);

const persistor = persistStore(store);

export {
  store,
  persistor,
  middleware
};



