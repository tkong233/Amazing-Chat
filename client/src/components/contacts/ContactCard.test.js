import React from 'react';
import ContactCard from './ContactCard';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16'; 
import { applyMiddleware, createStore } from 'redux';
import {middleware} from '../../store';
import rootReducer from '../../reducers/index'

Enzyme.configure({
    adapter: new EnzymeAdapter(),
});

const setUp = (initialState={}) =>{
    const user = {
        username: 'testuser',
        email: 'testemail',
        profile_picture: 'testpicture'
    };
    const contact = {
        contacts: [],
        suggestion: []
    };
    const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
    const store = createStoreWithMiddleware(rootReducer, initialState);
    const wrapper = shallow(<ContactCard store={store} user={user} contact={contact}/>).dive();
    // console.log(wrapper.debug());
    return wrapper;
}

describe('ContactCard Component', ()=>{
    describe('Checking PropTypes', ()=>{
        it('Should not throw a warning', ()=>{
            const expectedProps = {
                auth: {
                    user: {}
                },
                user: {},
            }
            // eslint-disable-next-line react/forbid-foreign-prop-types
            const propsErr = checkPropTypes(ContactCard.propTypes, expectedProps, 'props', ContactCard.name);
            expect(propsErr).toBeUndefined();
        })
    });
    describe('Renders', ()=>{
        let wrapper;
        beforeEach(()=>{
            const initialState = {
            };
            wrapper = setUp(initialState);
        });

        it('Should render without errors', ()=>{
            const container = wrapper.find(`[data-test='ContactCardComponent']`);
            expect(container.length).toBe(1);
        });

        it('Snapshot testing', ()=>{
            expect(wrapper).toMatchSnapshot();
        });


    });
});