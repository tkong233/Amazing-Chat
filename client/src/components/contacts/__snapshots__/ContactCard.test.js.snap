// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard Component Renders Snapshot testing 1`] = `ShallowWrapper {}`;
